### PR TITLE
Update app_signals to application_signals

### PR DESCRIPTION
**Pull request by bjrara**
*Issue #, if available:*

*Description of changes:*
Rename app_signals to application_signals for GA release. It's part of the PR: https://github.com/aws/amazon-cloudwatch-agent/pull/1133.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/instrumentation/defaultinstrumentation_test.go [hunk 1]
```diff
@@ -39,27 +39,29 @@ func Test_getDefaultInstrumentation(t *testing.T) {
 			Java: v1alpha1.Java{
 				Image: defaultJavaInstrumentationImage,
 				Env: []corev1.EnvVar{
-					{Name: "OTEL_SMP_ENABLED", Value: "true"}, //TODO: remove in favor of new name once safe
-					{Name: "OTEL_AWS_APP_SIGNALS_ENABLED", Value: "true"},
+					{Name: "OTEL_AWS_APP_SIGNALS_ENABLED", Value: "true"}, //TODO: remove in favor of new name once safe
```
Comment (okankoAMZ) [inline pkg/instrumentation/defaultinstrumentation_test.go:42]: Do we still need this comment?

File: pkg/instrumentation/podmutator_test.go [hunk 2]
```diff
@@ -51,27 +51,29 @@ func TestGetInstrumentationInstanceFromNameSpaceDefault(t *testing.T) {
 			Java: v1alpha1.Java{
 				Image: defaultJavaInstrumentationImage,
 				Env: []corev1.EnvVar{
-					{Name: "OTEL_SMP_ENABLED", Value: "true"}, //TODO: remove in favor of new name once safe
-					{Name: "OTEL_AWS_APP_SIGNALS_ENABLED", Value: "true"},
+					{Name: "OTEL_AWS_APP_SIGNALS_ENABLED", Value: "true"}, //TODO: remove in favor of new name once safe
```
Comment (okankoAMZ) [inline pkg/instrumentation/podmutator_test.go:54]: Same here, do we need this comment?

File: pkg/instrumentation/podmutator_test.go [hunk 3]
```diff
@@ -51,27 +51,29 @@ func TestGetInstrumentationInstanceFromNameSpaceDefault(t *testing.T) {
...
+					{Name: "OTEL_AWS_APPLICATION_SIGNALS_ENABLED", Value: "true"},
 					{Name: "OTEL_TRACES_SAMPLER_ARG", Value: "endpoint=http://cloudwatch-agent.amazon-cloudwatch:2000"},
 					{Name: "OTEL_TRACES_SAMPLER", Value: "xray"},
 					{Name: "OTEL_EXPORTER_OTLP_PROTOCOL", Value: "http/protobuf"},
 					{Name: "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT", Value: "http://cloudwatch-agent.amazon-cloudwatch:4316/v1/traces"},
-					{Name: "OTEL_AWS_SMP_EXPORTER_ENDPOINT", Value: "http://cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"}, //TODO: remove in favor of new name once safe
-					{Name: "OTEL_AWS_APP_SIGNALS_EXPORTER_ENDPOINT", Value: "http://cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"},
+					{Name: "OTEL_AWS_APP_SIGNALS_EXPORTER_ENDPOINT", Value: "http://cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"}, //TODO: remove in favor of new name once safe
```
Comment (okankoAMZ) [inline pkg/instrumentation/podmutator_test.go:60]: Same here, do we need this comment?

File: pkg/instrumentation/defaultinstrumentation_test.go [hunk 4]
```diff
@@ -88,27 +90,29 @@ func Test_getDefaultInstrumentation(t *testing.T) {
...
+					{Name: "OTEL_AWS_APPLICATION_SIGNALS_ENABLED", Value: "true"},
 					{Name: "OTEL_TRACES_SAMPLER_ARG", Value: "endpoint=http://cloudwatch-agent.amazon-cloudwatch:2000"},
 					{Name: "OTEL_TRACES_SAMPLER", Value: "xray"},
 					{Name: "OTEL_EXPORTER_OTLP_PROTOCOL", Value: "http/protobuf"},
 					{Name: "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT", Value: "https://cloudwatch-agent.amazon-cloudwatch:4316/v1/traces"},
-					{Name: "OTEL_AWS_SMP_EXPORTER_ENDPOINT", Value: "https://cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"}, //TODO: remove in favor of new name once safe
-					{Name: "OTEL_AWS_APP_SIGNALS_EXPORTER_ENDPOINT", Value: "https://cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"},
+					{Name: "OTEL_AWS_APP_SIGNALS_EXPORTER_ENDPOINT", Value: "https://cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"}, //TODO: remove in favor of new name once safe
```
Comment (okankoAMZ) [inline pkg/instrumentation/defaultinstrumentation_test.go:99]: Same here, do we need this comment?

File: pkg/instrumentation/defaultinstrumentation_test.go [hunk 5]
```diff
@@ -88,27 +90,29 @@ func Test_getDefaultInstrumentation(t *testing.T) {
 			Java: v1alpha1.Java{
 				Image: defaultJavaInstrumentationImage,
 				Env: []corev1.EnvVar{
-					{Name: "OTEL_SMP_ENABLED", Value: "true"}, //TODO: remove in favor of new name once safe
-					{Name: "OTEL_AWS_APP_SIGNALS_ENABLED", Value: "true"},
+					{Name: "OTEL_AWS_APP_SIGNALS_ENABLED", Value: "true"}, //TODO: remove in favor of new name once safe
```
Comment (okankoAMZ) [inline pkg/instrumentation/defaultinstrumentation_test.go:93]: Same here, do we need this comment?

File: pkg/instrumentation/defaultinstrumentation_test.go [hunk 6]
```diff
@@ -39,27 +39,29 @@ func Test_getDefaultInstrumentation(t *testing.T) {
...
+					{Name: "OTEL_AWS_APPLICATION_SIGNALS_ENABLED", Value: "true"},
 					{Name: "OTEL_TRACES_SAMPLER_ARG", Value: "endpoint=http://cloudwatch-agent.amazon-cloudwatch:2000"},
 					{Name: "OTEL_TRACES_SAMPLER", Value: "xray"},
 					{Name: "OTEL_EXPORTER_OTLP_PROTOCOL", Value: "http/protobuf"},
 					{Name: "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT", Value: "http://cloudwatch-agent.amazon-cloudwatch:4316/v1/traces"},
-					{Name: "OTEL_AWS_SMP_EXPORTER_ENDPOINT", Value: "http://cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"}, //TODO: remove in favor of new name once safe
-					{Name: "OTEL_AWS_APP_SIGNALS_EXPORTER_ENDPOINT", Value: "http://cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"},
+					{Name: "OTEL_AWS_APP_SIGNALS_EXPORTER_ENDPOINT", Value: "http://cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"}, //TODO: remove in favor of new name once safe
```
Comment (okankoAMZ) [inline pkg/instrumentation/defaultinstrumentation_test.go:48]: Same here, do we need this comment?

File: pkg/instrumentation/defaultinstrumentation.go [hunk 7]
```diff
@@ -63,27 +63,29 @@ func getDefaultInstrumentation(agentConfig *adapters.CwaConfig) (*v1alpha1.Instr
 			Java: v1alpha1.Java{
 				Image: javaInstrumentationImage,
 				Env: []corev1.EnvVar{
-					{Name: "OTEL_SMP_ENABLED", Value: "true"}, //TODO: remove in favor of new name once safe
-					{Name: "OTEL_AWS_APP_SIGNALS_ENABLED", Value: "true"},
+					{Name: "OTEL_AWS_APP_SIGNALS_ENABLED", Value: "true"}, //TODO: remove in favor of new name once safe
```
Comment (okankoAMZ) [inline pkg/instrumentation/defaultinstrumentation.go:66]: Same here, do we need this comment?

File: pkg/instrumentation/defaultinstrumentation.go [hunk 8]
```diff
@@ -63,27 +63,29 @@ func getDefaultInstrumentation(agentConfig *adapters.CwaConfig) (*v1alpha1.Instr
...
+					{Name: "OTEL_AWS_APPLICATION_SIGNALS_ENABLED", Value: "true"},
 					{Name: "OTEL_TRACES_SAMPLER_ARG", Value: "endpoint=http://cloudwatch-agent.amazon-cloudwatch:2000"},
 					{Name: "OTEL_TRACES_SAMPLER", Value: "xray"},
 					{Name: "OTEL_EXPORTER_OTLP_PROTOCOL", Value: "http/protobuf"},
 					{Name: "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT", Value: exporterPrefix + "cloudwatch-agent.amazon-cloudwatch:4316/v1/traces"},
-					{Name: "OTEL_AWS_SMP_EXPORTER_ENDPOINT", Value: exporterPrefix + "cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"}, //TODO: remove in favor of new name once safe
-					{Name: "OTEL_AWS_APP_SIGNALS_EXPORTER_ENDPOINT", Value: exporterPrefix + "cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"},
+					{Name: "OTEL_AWS_APP_SIGNALS_EXPORTER_ENDPOINT", Value: exporterPrefix + "cloudwatch-agent.amazon-cloudwatch:4316/v1/metrics"}, //TODO: remove in favor of new name once safe
```
Comment (okankoAMZ) [inline pkg/instrumentation/defaultinstrumentation.go:72]: Same here, do we need this comment?

**Review by okankoAMZ — APPROVED** [review]
Looks good but I am not sure if need those comments

File: pkg/instrumentation/defaultinstrumentation_test.go (hunk 1 above)
Comment (bjrara) [reply pkg/instrumentation/defaultinstrumentation_test.go:42]: Yes, all `AWS_APP_SIGNALS` will be replaced by `AWS_APPLICATION_SIGNALS`

**Comment by bjrara** [comment]
All `AWS_APP_SIGNALS` will be replaced by `AWS_APPLICATION_SIGNALS`.

File: pkg/instrumentation/defaultinstrumentation.go (hunk 7 above)
Comment (lisguo) [reply pkg/instrumentation/defaultinstrumentation.go:66]: Think it exists so that we are backwards compatible until the SDK releases the new version

**Review by lisguo — APPROVED** [review]

**Pull request merged**